src/lib/responses/error.response.ts

```diff
@@ -0,0 +1,22 @@
+import { Response } from "express";
+
+interface ErrorInterface {
+    error: string,
+    statusCode: number,
+    timestamp: number
+}
+
+function errorResponse(res: Response, statusCode: number, message: string): void {
+    const errorMessage = {
+        error: message,
+        statusCode: statusCode,
+        timestamp: Date.now()
+    } as ErrorInterface;
+
+    res.status(statusCode)
```

Comment:
we actually already have some error handling logic in `index.ts`
now we're adding additional error handling in a new middleware

I think that we should remove the error handling currently present in the restToNatsBridge and add a new error handling middlware, added at the end of the middleware stack.

See this for further information: https://expressjs.com/en/guide/error-handling.html

how we should handle errors:

1. if it's a client error, we should return the appropriate error code (4xx):
 - most client error like incorrect request payload would be 400
 - not authenticated is 401 as you already have it
 - not authorized is 403 as you already have it
 - endpoint not found is 404 (we have that logic in the restToNats middleware)
 - we should also log the error with `console.error`  (structured logging will come very soon so just a string is fine for now)

2. if it's a server error, like a bug in our code, then
 - return a 500 error with a generic message like `server error`
 - console.error the error, making sure it includes the stack trace

all of that logic should be put into one common error handling middleware

Reply:
what you can do is creating new Error types that extend the base error. each error type would have its own equivalent http error code. In the error handler you can then just reuse that code to add to the `res.status`, and if you want to do additional handling (like stopping the process if something bad happened) you can then do something like `if (e instance of ReallyBadError) { process.exit(1) }`

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error

Reply:
Agree, instead of checking each error type, could we have an object for each kind of error that return a pre-formatted object, such as :
`
function UnauthorizedRequestError(message: string, stack: string) : ResponseErrorInterface {
    return {
        name: "Unauthorized",
        stack: stack,
        statusCode: 403,
        message: message,
    }
}
`

And therefore our error middleware could look like :
`
function errorHandlingMiddleware(error: ResponseErrorInterface, req: Request, res: Response, next: NextFunction) : Response {
    return res.status(error.statusCode).send(error)
}
`

Reply:
Also added in natsToBridge. 
I left the error code based error handling in natsToBridge in order to give a better insight on the error, but we could also directly send the catched error in next() to manage this in the error middleware.